chore: BROS-194: Small code improvements

hlomzik opened the pull request:
Just a preparation for https://github.com/HumanSignal/label-studio/pull/8108 to keep the main PR clean.

### Annotation
- `forEach()` -> `for .. of` conversions.
  MST maps were not touched except for `keys()` access which is easier with `for .. of`.
- Bunch of optional chaining added to simplify code
- Removed unused `_renderAll()` method and `Segment` component

### Regions and Results
- Region/Result's `unselectRegion()` and connected code were not used.
- The whole `ImageRegion` is not in use.

### NodeView
- We have no usages of functional `altIcon`, so this wrapper is excess.

hlomzik committed:
Remove some unused code

Region/Result's `unselectRegion()` and connected code.
The whole `ImageRegion` is not in use.

hlomzik committed:
Small code improvements, more like a linting

- `forEach()` -> `for .. of` conversions.
  MST maps were not touched except for `keys()` access which is easier with `for .. of`.
- Bunch of optional chaining added to simplify code
- `Number.isNaN()` instead of `isNaN()`

hlomzik committed:
Simplify Node view, remove NodeView

We have no usages of functional `altIcon`, this wrapper is excess

hlomzik committed:
`Number.isNaN()` and `isNaN()` are not identical

We need to check whether it can be converted to number, so we need `isNaN()`.
https://stackoverflow.com/questions/33164725

nick-skriabin commented on web/libs/editor/src/stores/Annotation/Annotation.js:
```diff
@@ -1222,21 +1219,24 @@ const _Annotation = types
+          for (const r of self.results.filter((r) => r.area.classification)) {
+            r.from_name.updateFromResult?.(r.mainValue);
+          }
+        }
```
if i'm not mistaken, filter will run on each iteration. it's safer and cleaner to move it to a separate variable

```suggestion
        if (!hidden) {
          const filteredResults = self.results.filter((r) => r.area.classification); 
          for (const r of filteredResults) {
            r.from_name.updateFromResult?.(r.mainValue);
          }
        }
```

hlomzik committed:
Remove unused `_renderAll()` and `Segment` component

The main usage of this component was in `_renderAll()` and we don't use it
anymore.
The only usage left is in old UI which will be removed soon, so we basically
don't use `Segment` component.

hlomzik committed:
Fix filtering

moving to separate var for better performance

Co-authored-by: Nick Skriabin <767890+nick-skriabin@users.noreply.github.com>

hlomzik committed:
Improve filtering and revert some iterations

hlomzik committed:
Remove unused Tree

hlomzik committed:
`self.root` can be not defined, so secure `beforeSend()`

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/16940694748)
> Successfully merged:  create mode 100644 web/libs/editor/src/lib/AudioUltra/Visual/Renderer/ResizeRenderer.ts

hlomzik committed:
Merge branch 'develop' into 'fb-bros-194/preparation'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/16940694748
